<launch>
    <!-- Files to launch on static workstation for interfacing and monitoring robot -->
    <arg name="goalnav_on" default="false"/>
    <arg name="teleop_on" default="true"/>
    <arg name="joystick_on" default="false"/>
    <arg name="amcl_on" default="false"/>
    <arg name="slam_on" default="true"/>

    <!-- RQT Teleop Controller -->
    <group if="$(arg teleop_on)">
        
        <node name="rqt_drive" pkg="rqt_robot_steering" type="rqt_robot_steering" >
            <remap from="cmd_vel" to="/ottobot_diff_drive_controller/cmd_vel" />
        </node>
    </group>
    <!-- JOYSTICK using xbox controller-->
    <group if="$(arg joystick_on)">  
        <param name="joy_node/dev" value="/dev/input/js0" />
        <node name="joystick" pkg="joy" type="joy_node">
            <param name="autorepeat_rate" value="20" />
        </node>
        <node name="joystick_controller" pkg="teleop_twist_joy" type="teleop_node">
            <rosparam file="$(find ottobot_workstation)/config/joystick_params.yaml" command="load" />
            <remap from="cmd_vel" to="/ottobot_diff_drive_controller/cmd_vel" />
        </node>
    </group>

    <!--launch rviz-->
    <group if="$(eval amcl_on or slam_on)">
        <node name="rviz" pkg="rviz" type="rviz" respawn="false" args="-d $(find ottobot_robot)/rviz/slam_config.rviz"/>
    </group>
    <group unless="$(eval amcl_on or slam_on)">
        <node name="rviz" pkg="rviz" type="rviz" respawn="false" args="-d $(find ottobot_robot)/rviz/teleop_config.rviz"/>
    </group>

</launch>